src/Text/Markdown/SlamDown/Eval.purs

```diff
@@ -62,19 +62,19 @@ eval fs = everywhereM b i
       quoteTextBox = SD.transTextBox (runIdentity >>> SD.Literal >>> M.Just >>> Compose)
 
   f (SD.RadioButtons sel opts) = do
-    sel' <- evalExpr fs.value sel
     opts' <- evalExpr fs.list opts
-    pure $ SD.RadioButtons sel' (mergeSelection sel' opts')
+    sel' ← evalExpr (fs.value >=> \x → pure $ M.fromMaybe 0 $ L.elemIndex x (getValues opts')) sel
```

Comment:
Hmm, this is going to be fairly gnarly.

Now that I think of it, the logic in this is pretty bizarre. We have a `Expr Int`, but we are in fact expecting it to compute a *value* (`a`); so, the user has to write code that computes a value of type `a`, and then we run it to an integer / index. This weirdness isn't new: it's also present for checkboxes, via that `List Boolean` representation.

I don't know that it is worth addressing, since it would take serious engineering time to clean it up and we don't really care about Markdown in the distant future anyway.

Frankly, I think the big problem here is that we designed this library to be WAY too modular—and then we tried to use it with SlamData where we had very specific use-cases. Then, we ended up subverting the modularity in order to support these use-cases, in such a way that it superficially appears to be modular, but in fact makes a lot of SD-specific assumptions under the hood.

@garyb